phpBB/phpbb/extension/extension_interface.php

```diff
@@ -27,6 +27,13 @@
 	public function is_enableable();
 
 	/**
+	* Indicate why the extension can not be enabled.
+	*
+	* @return string|array with actual processed language string(s) (not language keys), or empty
+	*/
+	public function not_enableable_reason();
```

Comment:
That would be a tremendous backward compatibility break: a lot of the currently approved extensions would stop working, and a lot of extensions in development, or even abandoned, would fail for this single reason.
It would have been better that way at the very beginning, but not now, in my view.  The intention here was to keep the current interface intact, and that all existing extensions would continue working without any change.  And if some extension wants to change, then it is easy to do it this way, while keeping 100% BC.